src/module/apps/compendium-browser/components/filters.svelte

```diff
@@ -0,0 +1,185 @@
+            <label id="sort-order">
+                {game.i18n.localize("PF2E.CompendiumBrowser.Filter.OrderByLabel")}:
+                <div class="select-container">
+                    <select bind:value={filter.order.by} onchange={onChangeSortValue}>
+                        {#each R.entries(filter.order.options) as [key, data]}
+                            <option value={key}>{game.i18n.localize(data.label)}</option>
+                        {/each}
+                    </select>
+                    <button onclick={onChangeSortOrder} aria-labelledby="sort-order" type="button" class="order-button">
+                        {#if filter.order.type === "alpha"}
+                            {#if filter.order.direction === "asc"}
+                                <i class="fa-solid fa-sort-alpha-up"></i>
+                            {:else}
+                                <i class="fa-solid fa-sort-alpha-down-alt"></i>
+                            {/if}
+                        {:else if filter.order.type === "numeric"}
+                            {#if filter.order.direction === "asc"}
+                                <i class="fa-solid fa-sort-numeric-up"></i>
+                            {:else}
+                                <i class="fa-solid fa-sort-numeric-down-alt"></i>
+                            {/if}
+                        {/if}
+                    </button>
+                </div>
+            </label>
+            {#if "selects" in filter}
+                {#each R.entries(filter.selects) as [key, data]}
+                    <label>
+                        {game.i18n.localize(data.label)}:
+                        <div class="select-container">
+                            <select bind:value={filter.selects[key].selected} data-key={key}>
+                                <option value="">-</option>
+                                {#each R.entries(data.options) as [key, label]}
+                                    <option value={key}>{game.i18n.localize(label)}</option>
+                                {/each}
+                            </select>
+                        </div>
+                    </label>
+                {/each}
+            {/if}
+        </div>
+        <button type="button" class="clear-filters" onclick={() => resetFilters()}>
+            {game.i18n.localize("PF2E.CompendiumBrowser.Filter.ClearAllFilters")}
+        </button>
+    </div>
+    <FilterContainer label="PF2E.Traits">
+        <Traits bind:traits={filter.traits} />
+    </FilterContainer>
+    {#each Object.entries(filter.checkboxes) as [key, checkbox]}
+        <FilterContainer
+            isExpanded={checkbox.isExpanded}
+            clearButton={{ data: checkbox, options: { visible: checkbox.selected.length > 0 } }}
+            label={checkbox.label}
+        >
+            <Checkboxes bind:checkbox={filter.checkboxes[key as keyof BrowserFilter["checkboxes"]]} />
+        </FilterContainer>
+    {/each}
```

Comment:
`R.entries` adds a`never[]` here for some reason so I resorted to type assertion. A union type of all checkbox filters does not include `never` so I'm not really sure what's going on there.

Reply:
I tested with converting everything to Record<"string" | "unions", CheckboxData> but that doesn't seem to work at all. Seems to be an issue with R.entries's implementation on a union of records with explicit keys.